From 3a08ca094422dc873d9c48fda20ddf177d28f432 Mon Sep 17 00:00:00 2001
From: Jan Kratochvil <jan.kratochvil@redhat.com>
Date: Wed, 2 May 2012 20:00:31 +0000
Subject: [PATCH] gdb/
 	Fix --without-auto-load-safe-path for MS-Windows host platform.
 	* auto-load.c (filename_is_in_dir): Return 1 for DIR_LEN 0.

---
 gdb/ChangeLog   |    5 +++++
 gdb/auto-load.c |    6 ++++++
 2 files changed, 11 insertions(+), 0 deletions(-)

Index: gdb-7.2/gdb/auto-load.c
===================================================================
--- gdb-7.2.orig/gdb/auto-load.c
+++ gdb-7.2/gdb/auto-load.c
@@ -231,6 +231,12 @@ filename_is_in_dir (const char *filename
   while (dir_len && IS_DIR_SEPARATOR (dir[dir_len - 1]))
     dir_len--;
 
+  /* Ensure auto_load_safe_path "/" matches any FILENAME.  On MS-Windows
+     platform FILENAME even after gdb_realpath does not have to start with
+     IS_DIR_SEPARATOR character, such as the 'C:\x.exe' filename.  */
+  if (dir_len == 0)
+    return 1;
+
   return (filename_ncmp (dir, filename, dir_len) == 0
 	  && (IS_DIR_SEPARATOR (filename[dir_len])
 	      || filename[dir_len] == '\0'));
